superset-frontend/src/views/CRUD/data/dataset/DatasetLayout/DatasetLayout.test.tsx

```diff
@@ -33,10 +33,21 @@ describe('DatasetLayout', () => {
     expect(layoutWrapper).toHaveTextContent('');
   });
 
-  it('renders a Header when passed in', () => {
-    render(<DatasetLayout header={Header()} />);
+  const mockSetDataset = jest.fn();
 
-    expect(screen.getByText(/header/i)).toBeVisible();
+  const waitForRender = () =>
+    waitFor(() =>
+      render(<Header setDataset={mockSetDataset} datasetName="" />),
```

Comment:
nit: add a test to make sure if `datasetName` is set that value of the textbox renders